Following error was occured when testing the connector with client credential grant 
```
[2024-02-19 12:06:19,029] [138f7e49-2aba-4626-bd52-4112e2a7ed0e] ERROR {org.wso2.carbon.identity.provisioning.ProvisioningThread} -  Provisioning for Entity sample@gmail.com For operation = POST org.wso2.carbon.identity.provisioning.IdentityProvisioningException: Error in invoking provisioning operation for the user
	at org.wso2.carbon.identity.provisioning.connector.salesforce.SalesforceProvisioningConnector.createUser(SalesforceProvisioningConnector.java:261)
	at org.wso2.carbon.identity.provisioning.connector.salesforce.SalesforceProvisioningConnector.provision(SalesforceProvisioningConnector.java:105)
	at org.wso2.carbon.identity.provisioning.ProvisioningThread.call(ProvisioningThread.java:85)
	at org.wso2.carbon.identity.provisioning.ProvisioningThread.call(ProvisioningThread.java:34)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.net.ssl.SSLPeerUnverifiedException: Certificate for <wso243-dev-ed.develop.my.salesforce.com> doesn't match any of the subject alternative names: [a.sfdc-y37hzm.edge.salesforce.com, *.my.salesforce.com, *.force.com, *.d.forceusercontent.com, *.c.forceusercontent.com, *.livepreview.salesforce-communities.com, *.cloudforce.com, *.builder.salesforce-communities.com, *.database.com, *.a.forceusercontent.com, *.sandbox.force.com, *.lightning.force.com, *.documentforce.com, *.secure.force.com, *.preview.salesforce-communities.com, *.visualforce.com, *.container.lightning.com, *.b.forceusercontent.com]
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.verifyHostname(SSLConnectionSocketFactory.java:507)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:437)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at org.wso2.carbon.identity.provisioning.connector.salesforce.SalesforceProvisioningConnector.createUser(SalesforceProvisioningConnector.java:234)
	... 7 more

```
But If we note that we were able to get access token and the certificate call fails at user creation.  So this is a deployment specific issues not the issue raised by the change prposed

--- a/components/org.wso2.carbon.identity.provisioning.connector.salesforce/src/main/java/org/wso2/carbon/identity/provisioning/connector/salesforce/SalesforceProvisioningConnectorFactory.java
+++ b/components/org.wso2.carbon.identity.provisioning.connector.salesforce/src/main/java/org/wso2/carbon/identity/provisioning/connector/salesforce/SalesforceProvisioningConnectorFactory.java
@@ -91,12 +91,20 @@ public class SalesforceProvisioningConnectorFactory extends AbstractProvisioning
         clientSecret.setConfidential(true);
         configProperties.add(clientSecret);
 
+        Property grantType = new Property();
+        grantType.setName(SalesforceConnectorConstants.PropertyConfig.USE_PASSWORD_GRANT);
+        grantType.setDisplayName("Use OAuth Username-Password Flow to get access token");
+        grantType.setType("boolean");
+        grantType.setDefaultValue("true");
+        grantType.setDisplayOrder(5);
+        configProperties.add(grantType);
+
         Property username = new Property();
         username.setName(SalesforceConnectorConstants.PropertyConfig.USERNAME);
         username.setDisplayName("Username");
         username.setRequired(false);
         username.setType("string");
-        username.setDisplayOrder(5);
+        username.setDisplayOrder(6);
         configProperties.add(username);
 
         Property password = new Property();
@@ -104,17 +112,9 @@ public class SalesforceProvisioningConnectorFactory extends AbstractProvisioning
         password.setDisplayName("Password");
         password.setRequired(false);
         password.setType("string");
-        password.setDisplayOrder(6);
+        password.setDisplayOrder(7);
         password.setConfidential(true);
         configProperties.add(password);
-
-        Property grantType = new Property();
-        grantType.setName(SalesforceConnectorConstants.PropertyConfig.USE_PASSWORD_GRANT);
-        grantType.setDisplayName("Use Username-Password grant to get access token");
-        grantType.setType("boolean");
-        grantType.setDefaultValue("true");
-        grantType.setDisplayOrder(7);
-        configProperties.add(grantType);
 
         Property tokenEp = new Property();
         tokenEp.setName(SalesforceConnectorConstants.PropertyConfig.OAUTH2_TOKEN_ENDPOINT);

